### PR TITLE
Adding the event moved in send_headers

### PR DESCRIPTION
**Pull request by PrimozGodec**
**Issue:** Since adding to the dict was after awaitable `send_headers` it can happen that the same event is pooled from events in the `receive_event` before it is added to the dictionary. In that case, the program fails.

**Fix:** `self.events[stream_id] = []` was moved in function `send_headers` to avoid this situation.

### CODE REVIEW

File: httpx/dispatch/http2.py [hunk 1]
```diff
@@ -48,7 +48,6 @@ def __init__(
 
         stream_id = await self.send_headers(request, timeout)
 
-        self.events[stream_id] = []
```
[the remark this answers is not in the document]
Comment (PrimozGodec) [reply httpx/dispatch/http2.py]: That is actually a better option. I change that.

**Comment by PrimozGodec** [comment]
@florimondmanca I think it is worth a test case but I do not have an idea for an elegant solution right now, except sending many requests and see what happens.

**Comment by tomchristie** [comment]
Thanks for having submitted this. Looks to me like it's now obsolete. Apologies for not having gotten to it sooner.

**Comment by PrimozGodec** [comment]
No problem. I can confirm that it is obsolete.

**Pull request closed without merging**